### PR TITLE
[variations] Support multiple axes with same tag, aka HOI

### PR DESCRIPTION
**Pull request by behdad**
Fixes https://github.com/harfbuzz/harfbuzz/issues/1673
Fixes https://github.com/harfbuzz/harfbuzz/issues/2743

Test from https://github.com/ctrlcctrlv/FontForge-Higher-Order-Interpolation

### CODE REVIEW

**Comment by behdad** [comment]
cc @bungeman @lemzwerg @justvanrossum @ctrlcctrlv

**Comment by lemzwerg** [comment]
I don't know enough about HarfBuzz to make any qualified review, but I'm all for it, so thanks for working on this!

Is there anything necessary on the FreeType side?

File: src/hb-font.cc [hunk 1]
```diff
@@ -2067,13 +2067,16 @@ hb_font_set_variations (hb_font_t            *font,
   const OT::fvar &fvar = *font->face->table.fvar;
   for (unsigned int i = 0; i < variations_length; i++)
   {
-    unsigned axis_index;
-    if (fvar.find_axis_index (variations[i].tag, &axis_index) &&
-	axis_index < coords_length)
+    unsigned axis_index_start, axis_index_end;
+    if (fvar.find_axis_index (variations[i].tag, &axis_index_start, &axis_index_end))
```
Comment (drott) [inline src/hb-font.cc:2071]: If there are multiple axes with the same tag, is it ensured that they are ordered and identical tags are lined up consecutively? In the font, or does HB sort them somewhere?

Comment (drott) [reply src/hb-font.cc:2071]: PS: If yes, then LGTM.

Comment (justvanrossum) [reply src/hb-font.cc:2071]: I don't think the OT spec demands the axes to be ordered by tag, and `find_axis_index ` seems to iterate using the order in the font: I suspect that `find_axis_index()` is not correct.

Comment (behdad) [reply src/hb-font.cc:2071]: Ah, I assumed they are sorted by tag. Let me check.

Comment (behdad) [reply src/hb-font.cc:2071]: Okay yes, not sorted. I'll rework the code.

Comment (behdad) [reply src/hb-font.cc]: I pushed a reworked version that doesn't make sorted assumption. PTAL.

**Comment by behdad** [comment]
> Is there anything necessary on the FreeType side?

Not that I know of. From what I've heard, FreeType works fine. Thanks.

**Review by drott — APPROVED** [review]
LGTM with this change - although I have limited knowledge of this code.

File: src/hb-font.cc (hunk 1 above)
Comment (justvanrossum) [reply src/hb-font.cc]: Looks good to me.

**Pull request merged**